connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -758,9 +758,11 @@ public void run() {
     }
 
     // Common handling for requests that get config data. Checks if we are in sync with the current config, which allows
-    // us to answer requests directly. If we are not, handles invoking the callback with the appropriate error.
+    // us to answer requests directly. If we are not, handles invoking the callback with the appropriate error;
+    // Note that the configState's offset can be bigger than assignment's offset; Because if updating an existing
+    // connector, the connector can pick up the config change without a rebalance;
     private boolean checkConfigSynced(Callback<?> callback) {
-        if (assignment == null || configState.offset() != assignment.offset()) {
+        if (assignment == null || configState.offset() < assignment.offset()) {
```

Comment:
@ewencp 
Thanks for comment;
in DistributedHerder,  it seems that checkConfigSynced is only used in:
```java
public synchronized void connectors(final Callback<Collection<String>> callback)
public synchronized void connectorInfo(final String connName, final Callback<ConnectorInfo> callback)
public synchronized void taskConfigs(final String connName, final Callback<List<TaskInfo>> callback)
```
Above methods are only used for querying information or config for connector or tasks;
Is it really necessary to check if configState.offset equals to assignment.offset?
Or can we remove this check and just return information generated from configState? It contains config being used currently;
If user want to know if the config is synced, we can return the status, which tell if offset of configState is bigger, smaller than or equals to assignment's offset;